File: Problem/BOJ_1181_단어 정렬/main.py
from sys import stdin


def main():
    stdin = open('./input.txt')
    num_of_words = int(stdin.readline())
    words = {}

    for _ in range(num_of_words):
        word = stdin.readline().strip()
        words[word] = len(word)

    words = sorted(words.items(), key=lambda x: (x[1], x[0]))

    for word, length in words:
        print(word, len(word))


if __name__ == '__main__':
    main()

File: Problem/BOJ_1181_단어 정렬/main.js
const fs = require('fs')
const input = fs.readFileSync("./input.txt").toString().trim().split("\n")

const numOfWords = parseInt(input[0])

const words = new Map()
input.slice(1).forEach(word => {
    words.set(word, word.length)
})


const sortedWords = [...words].sort((a, b) => {
    if (a[1] < b[1]) {
        return -1
    } else if (a[1] > b[1]) {
        return 1
    } else if (a[1] === b[1]) {
        return a[0].localeCompare(b[0])
    }
})

for (let i = 0; i < sortedWords.length; i++) {
    console.log(sortedWords[i][0])
}
